transforms/comp_visual_preset.py

```diff
@@ -0,0 +1,120 @@
+"""It's a visual preset consisting of a setup of color_correction, env_fog_controller, and env_tonemap_controller.
+"""
+
+from hammeraddons.bsp_transform import trans, Context
+from srctools.logger import get_logger
+from srctools import Entity, VMF, Output
+
+LOGGER = get_logger(__name__)
+
+
+
+@trans("comp_visual_preset")
+def visual_preset(ctx: Context) -> None:
+    """Sets a visual preset..."""
+    vpreset: Entity
+    vmf: VMF = ctx.vmf
+    for vpreset in vmf.by_class["comp_visual_preset"]:
+        vpreset.remove()
+        LOGGER.log(0, f"Parsing visual preset: {vpreset['targetname']}")
+
+        relay_ent = vmf.create_ent("logic_relay", 
+                       targetname = "visual_preset_" + vpreset["targetname"],
+                       angles = "0 0 0",
+                       spawnflags = 0,
+                       startdisabled = 0,
+                       origin = vpreset.get_origin()
+                       )
+        
+        
+        ctx.add_io_remap( # Rebind the IO
+                vpreset["targetname"],
+                Output("Apply", relay_ent, "Trigger")
+            )
+
+        # Tonemapping
+        tonemapper: Entity
+        to_tm_outputs = []
+        if (name := vpreset["tonemapper", None]) and (tonemappers := list(vmf.search(name))): # Check if we have set tonemapper, if it exists and set it as a variable for us to use
+            if tonemappers:
+                tonemapper = tonemappers[0]
+
+            to_tm_outputs = [
+                Output("OnTrigger", tonemapper, "SetAutoExposureMax", param=str(vpreset["tm_autoexposuremax", 2.0]) ),
+                Output("OnTrigger", tonemapper, "SetAutoExposureMin", param=str(vpreset["tm_autoexposuremin", 0.5]) ),
+                Output("OnTrigger", tonemapper, "SetBloomScale", param=str(vpreset["tm_bloomscale", 0.2]) ),
+                Output("OnTrigger", tonemapper, "SetBloomExponent", param=str(vpreset["tm_bloomexponent", 2.2]) ),
+            ]
+
+        relay_ent.add_out(*to_tm_outputs)
```

Comment:
Same as my prior comment. Also i noticed that it does create the tone mapping entity for you already. so maybe populating an ent is more consistent? unsure